Source/Private/ASLayoutTransition.mm

```diff
@@ -156,18 +180,40 @@ - (void)calculateSubnodeOperationsIfNeeded
   ASLayout *pendingLayout = _pendingLayout->layout;
 
   if (previousLayout) {
+#if AS_IG_LIST_KIT
+    // IGListDiff completes in linear time O(m+n), so use it if we have it:
+    IGListIndexSetResult *result = IGListDiff(previousLayout.sublayouts, pendingLayout.sublayouts, IGListDiffEquality);
+    _insertedSubnodePositions = findNodesInLayoutAtIndexes(pendingLayout, result.inserts, &_insertedSubnodes);
+    _removedSubnodePositions = findNodesInLayoutAtIndexes(previousLayout, result.deletes, &_removedSubnodes);
+    for (IGListMoveIndex *move in result.moves) {
+      unowned ASDisplayNode *subnode = previousLayout.sublayouts[move.from].layoutElement;
+      _subnodeMoves.push_back(std::pair<ASDisplayNode *, NSUInteger>(subnode, move.to));
+    }
+
+    // Sort by ascending order of move destinations, this will allow easy loop of `insertSubnode:AtIndex` later.
+    std::sort(_subnodeMoves.begin(), _subnodeMoves.end(), [](std::pair<id<ASLayoutElement>, NSUInteger> a,
+            std::pair<ASDisplayNode *, NSUInteger> b) {
+      return a.second < b.second;
+    });
+#else
     NSIndexSet *insertions, *deletions;
+    NSArray<NSIndexPath *> *moves;
     [previousLayout.sublayouts asdk_diffWithArray:pendingLayout.sublayouts
                                        insertions:&insertions
                                         deletions:&deletions
-                                     compareBlock:^BOOL(ASLayout *lhs, ASLayout *rhs) {
-                                       return ASObjectIsEqual(lhs.layoutElement, rhs.layoutElement);
-                                     }];
+                                            moves:&moves];
+
     _insertedSubnodePositions = findNodesInLayoutAtIndexes(pendingLayout, insertions, &_insertedSubnodes);
     _removedSubnodePositions = findNodesInLayoutAtIndexesWithFilteredNodes(previousLayout,
                                                                            deletions,
                                                                            _insertedSubnodes,
                                                                            &_removedSubnodes);
+    // These should arrive sorted in ascending order of move destinations.
+    for (NSIndexPath *move in moves) {
+      unowned ASDisplayNode *subnode = previousLayout.sublayouts[([move indexAtPosition:0])].layoutElement;
+      _subnodeMoves.push_back(std::pair<id, NSUInteger>(subnode, [move indexAtPosition:1]));
```

Comment:
- Is it valid to remove the parens on 213? `previousLayout.sublayouts[[move indexAtPosition:0]].layoutElement`
- Same `make_pair` game to safe space and retain/release traffic